gprofiler/spark/sampler.py

```diff
@@ -258,31 +61,31 @@ def _collect_loop(self) -> None:
         ), "A valid API client or storage directory is required"
         timefn = time.monotonic
         start_time = timefn()
+        discovered = False
+
         while not self._stop_event.is_set():
-            if self._spark_sampler is None:
-                spark_cluster_conf = self._find_spark_cluster()
-                if spark_cluster_conf is not None:
-                    master_address, cluster_mode = spark_cluster_conf
-                    self._spark_sampler = SparkCollector(cluster_mode, master_address)
-                else:
-                    if timefn() - start_time >= FIND_CLUSTER_TIMEOUT_SECS:
+            if self._big_data_sampler is not None:
```

Comment:
What is this check for? It is never None. You initialize it in the c'tor.

Reply:
f1410bab8ad0a9af9831f554578de4e02db80303